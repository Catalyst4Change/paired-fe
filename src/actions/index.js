export const setLoading = bool => ({
  type: 'SET_LOADING',
  bool
});

export const setError = error => ({
  type: 'SET_ERROR',
  error
});

export const setUser = user => ({
  type: 'SET_USER',
  user
});