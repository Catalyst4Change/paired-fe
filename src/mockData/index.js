export const mockUser = {
  name: "Hillary",
  program: "FE",
  module: 4,
  id: "5caa7eebfdebb8348e53a48e"
}

export const mockSchedule = [
  {
    pairer: {
      name: "Hillary",
      module: 4,
      program: "FE",
      id: "5caa7eebfdebb8348e53a48e",
      pronouns: "she/her",
      slack: "@hillstew"
    },
    pairee: null,
    date: "Mon Apr 15 2019",
    time: "lunch",
    id: "5caa7eebfdebb8348e53a490",
    notes: null
  },
  {
    pairer: {
      name: "Hillary",
      module: 4,
      program: "FE",
      id: "5caa7eebfdebb8348e53a48e",
      pronouns: "she/her",
      slack: "@hillstew"
    },
    paireee: {
      name: "Jeo",
      module: 4,
      program: "FE",
      pronouns: "he/him",
      slack: "@jeo",
    },
    date: "Wed Apr 17 2019",
    time: "afternoon",
    id: "5caa7eebfdebb8348e53a49d",
    notes: null,
  }
];

export const mockAvailPairings = [
  {
    pairer: {
      name: "Tiffany",
      module: 4,
      program: "FE",
      id: "5caa7eebfdebb8348e53a48d",
      pronouns: "she/her",
      skills: [
        "react",
        "testing"
      ],
      slack: "@tiffany",
    },
    date: "Thu Apr 18 2019",
    time: "morning",
    id: "5caa7eebfdebb8348e53a4a4",
  },
  {
    pairer: {
      name: "Tiffany",
      module: 4,
      program: "FE",
      id: "5caa7eebfdebb8348e53a48d",
      pronouns: "she/her",
      skills: [
        "react",
        "testing"
      ],
      slack: "@tiffany",
    },
    date: "Thu Apr 18 2019",
    time: "lunch",
    id: "5caa7eebfdebb8348e53a4a5",
  }
]; 

export const mockPairingToDeleteFromSched = {
  pairer: {
    name: 'Hillary',
    module: 4,
    program: 'FE',
    id: '5caa7eebfdebb8348e53a48e',
    pronouns: 'she/her',
    slack: '@hillstew'
  },
  pairee: null,
  date: 'Mon Apr 15 2019',
  time: 'lunch',
  id: '5caa7eebfdebb8348e53a490',
  notes: null
};

export const mockScheduleAfterDelete = [
  {
    pairer: {
      name: "Hillary",
      module: 4,
      program: "FE",
      id: "5caa7eebfdebb8348e53a48e",
      pronouns: "she/her",
      slack: "@hillstew"
    },
    paireee: {
      name: "Jeo",
      module: 4,
      program: "FE",
      pronouns: "he/him",
      slack: "@jeo",
    },
    date: "Wed Apr 17 2019",
    time: "afternoon",
    id: "5caa7eebfdebb8348e53a49d",
    notes: null,
  }
]; 

export const mockPairingToAddToSched = {
  pairer: {
    name: 'Hillary',
    module: 4,
    program: 'FE',
    id: '5caa7eebfdebb8348e53a48e',
    pronouns: 'she/her',
    slack: '@hillstew'
  },
  pairee: null,
  date: 'Thu Apr 18 2019',
  time: 'morning',
  id: '5caa7eebfdebb8348e56f000',
  notes: null
};

export const mockScheduleAfterAdd = [
  {
    pairer: {
      name: "Hillary",
      module: 4,
      program: "FE",
      id: "5caa7eebfdebb8348e53a48e",
      pronouns: "she/her",
      slack: "@hillstew"
    },
    pairee: null,
    date: "Mon Apr 15 2019",
    time: "lunch",
    id: "5caa7eebfdebb8348e53a490",
    notes: null
  },
  {
    pairer: {
      name: "Hillary",
      module: 4,
      program: "FE",
      id: "5caa7eebfdebb8348e53a48e",
      pronouns: "she/her",
      slack: "@hillstew"
    },
    paireee: {
      name: "Jeo",
      module: 4,
      program: "FE",
      pronouns: "he/him",
      slack: "@jeo",
    },
    date: "Wed Apr 17 2019",
    time: "afternoon",
    id: "5caa7eebfdebb8348e53a49d",
    notes: null,
  },
  mockPairingToAddToSched
];

export const mockQueryFromgetAvailablePairings = {
         query: `{
    getAvailablePairings(filter: { program: "BE", module: 3, date: "Wed Apr 17 2019" }) {
      pairer {
        name
        module
        program
        id
        pronouns
        skills
        slack
      }
      date
      time
      id
    }
  }`
       };

export const mockQueryFromgetUser = {
         query: `{
  getUser(name: "Hillary") {
    name
    program
    module
    id
  }
}`
       };

export const mockQueryFromgetUserPairings = {
         query: `{
    getUserPairings(
      id: "5h5h5h5h5h5h5"
    ) {
      pairer {
        name
        module
        program
        id
        pronouns
        slack
      }
      pairee {
        name
        module
        program
        pronouns
        slack
      }
      date
      time
      id
      notes
    }
  }`
       };

export const mockQueryFromupdatePairing = {
         query: `mutation {
    updatePairing(pairing: {
        id: "6t6t6t"
        pairee: "7b7b7b"
        notes: "Mythical Creatures"
    }) {
      pairer {
        name
        module
        program
        id
        pronouns
        slack
      }
      pairee {
        name
        module
        program
        id
        pronouns
        slack
      }
      date
      time
      id
      notes
    }
  }`
       };

export const mockQueryFromdeletePairing = {
         query: `mutation {
    deletePairing(id: "45dtl") {
      date
    }
  }`
       };