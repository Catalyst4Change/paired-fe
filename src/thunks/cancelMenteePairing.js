import { fetchData } from '../utils';
import { setError, setLoading } from '../actions';
import { getSchedule } from './getSchedule';
import * as gql from '../queries';

export const cancelMenteePairing = (pairingId, userId) => {
  return async dispatch => {
    try {
      dispatch(setLoading(true));
      const query = gql.cancelMenteePairing(pairingId);
      await fetchData(query);
      dispatch(setLoading(false));
      dispatch(getSchedule(userId));
    } catch (error) {
      dispatch(setLoading(false));
      dispatch(setError(error.message));
    }
  };
};
