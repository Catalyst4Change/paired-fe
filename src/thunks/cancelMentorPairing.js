import { fetchData } from '../utils';
import { setError, setLoading, cancelPairing } from '../actions';
import * as gql from '../queries';

export const cancelMentorPairing = (pairingId) => {
  return async dispatch => {
    try {
      dispatch(setLoading(true));
      const query = gql.cancelMentorPairing(pairingId);
      await fetchData(query);
      dispatch(setLoading(false));
      dispatch(cancelPairing(pairingId));
    } catch (error) {
      dispatch(setLoading(false));
      dispatch(setError(error.message));
    }
  };
};
