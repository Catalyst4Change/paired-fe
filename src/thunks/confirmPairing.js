import { fetchData } from '../utils';
import { setError, setLoading, addToSchedule, clearAvailPairings } from '../actions';
import * as gql from '../queries';

export const confirmPairing = (pairingId, paireeId, notes) => {
  return async dispatch => {
    dispatch(setLoading(true));
    try {
      const query = gql.updatePairing(pairingId, paireeId, notes);
      const response = await fetchData(query);
      const pairing = await response.updatePairing;
      dispatch(addToSchedule(pairing));
      dispatch(clearAvailPairings());
    } catch(error) {
      dispatch(setError(error.message));
    }
    dispatch(setLoading(false));
  }
};
