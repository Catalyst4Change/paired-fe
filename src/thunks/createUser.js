import { fetchData } from '../utils';
import { setError, setLoading, setUser } from '../actions';
import { getSchedule } from './getSchedule';
import * as gql from '../queries';
import * as helpers from '../helpers';

export const createUser = (userInfo, availabilities) => {
  return async dispatch => {
    dispatch(setLoading(true));
    try {
      const userQuery = gql.createUser(userInfo);
      const { user } = await fetchData(userQuery);
      const startDate = 'Mon Apr 29 2019';
      const inning = helpers.getDatesToDisplayFancy(startDate, 29);
      const unformattedPairings = helpers.createPairingsForQuery(
        availabilities,
        inning,
        user.id
      );
      const formattedPairings = helpers.formatPairingsForQuery(unformattedPairings)
      const pairingQuery = gql.createPairings(formattedPairings);
      await fetchData(pairingQuery);
      dispatch(setUser(user));
      dispatch(getSchedule(user.id));
    } catch (error) {
      dispatch(setError(error.message));
    }
    dispatch(setLoading(false));
  };
};




