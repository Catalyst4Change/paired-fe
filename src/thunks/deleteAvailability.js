import { fetchData } from "../utils";
import { setError, setLoading } from '../actions';
import * as gql from '../queries';

export const deleteAvailability = (userId) => {
  return async dispatch => {
    dispatch(setLoading(true));
    try {
      const mutation = gql.deletePairings(userId);
      await fetchData(mutation);
    } catch (error) {
      dispatch(setError(error.message));
    }
    dispatch(setLoading(false));
  }
};