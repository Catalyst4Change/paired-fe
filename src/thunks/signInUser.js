import { fetchData } from '../utils';
import { setError, setLoading, setUser } from '../actions';
import { getSchedule } from './getSchedule';
import * as gql from '../queries';

export const signInUser = (id) => {
  return async dispatch => {
    dispatch(setLoading(true));
    try {
      const userQuery = gql.getUserByFirebaseID(id);
      const { user } = await fetchData(userQuery);
      if (user !== null) {
        dispatch(setUser(user));
        dispatch(getSchedule(user.id));
      } else {
        dispatch(setUser({ isNewUser: true }));
      }
    } catch (error) {
      dispatch(setError(error.message));
    }
    dispatch(setLoading(false));
  };
};
