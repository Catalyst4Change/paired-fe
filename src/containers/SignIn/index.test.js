import React from 'react';
import { shallow } from 'enzyme';
import { mockUser } from '../../mockData';
import { SignIn, mapStateToProps, mapDispatchToProps } from './index';
import { setError } from '../../actions';
import { signInUser } from '../../thunks/signInUser';

jest.mock('../../thunks/signInUser');

describe('SignIn', () => {
  describe('SignIn component', () => {
    let wrapper;
    const mockSetError = jest.fn();
    const mockSignInUser = jest.fn();
    beforeEach(() => {
      wrapper = shallow(
        <SignIn
          setError={mockSetError}
          signInUser={mockSignInUser}
          user={{}}
        />
      );
    });

    it('should match the snapshot when there is no user', () => {
      expect(wrapper).toMatchSnapshot();
    });
    
    it('should match the snapshot when there is a new user', () => {
      wrapper = shallow(
        <SignIn
          setError={mockSetError}
          signInUser={mockSignInUser}
          user={{ isNewUser: true, firebaseID: 'abc123' }}
        />
      );
      expect(wrapper).toMatchSnapshot();
    });

    it('should match the snapshot when there is an existing user', () => {
      wrapper = shallow(
        <SignIn
          setError={mockSetError}
          signInUser={mockSignInUser}
          user={mockUser}
        />
      );
      expect(wrapper).toMatchSnapshot();
    });

    it('should call handleSignIn when the button is clicked', () => {
      jest.spyOn(wrapper.instance(), 'handleSignIn');
      wrapper.instance().forceUpdate();
      wrapper.find('button').simulate('click');
      expect(wrapper.instance().handleSignIn).toHaveBeenCalled();
    });
  });
});

describe('mapStateToProps', () => {
  it('should return a props object with the correct properties', () => {
    const initialState = { user: { id: 'abc123' }, extraProperty: true };
    const expected = { user: { id: 'abc123' } };
    const result = mapStateToProps(initialState);
    expect(result).toEqual(expected);
  });
});

describe('mapDispatchToProps', () => {
  const mockDispatch = jest.fn();
  const mappedProps = mapDispatchToProps(mockDispatch);

  it('should dispatch signInUser', () => {
    const expected = signInUser('abc123')
    mappedProps.signInUser('abc123');
    expect(mockDispatch).toHaveBeenCalledWith(expected);
  });
  
  it('should dispatch signUserOut', () => {
    const expected = setError('error signing in');
    mappedProps.setError('error signing in');
    expect(mockDispatch).toHaveBeenCalledWith(expected);
  });
});
