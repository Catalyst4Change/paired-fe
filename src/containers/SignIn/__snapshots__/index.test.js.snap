// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn SignIn component should match the snapshot 1`] = `
<div
  className="SignIn--div"
>
  <button
    onClick={[Function]}
  >
    Sign In with GitHub
  </button>
  <img
    alt="Two people coding"
    className="SignIn--img"
    src="code-typing.svg"
  />
</div>
`;
